src/TwigComponent/src/Command/TwigComponentDebugCommand.php

```diff
@@ -37,6 +36,7 @@ public function __construct(
         private string $twigTemplatesPath,
         private ComponentFactory $componentFactory,
         private Environment $twig,
+        private readonly ComponentPropertiesExtractor $componentPropertiesExtractor,
```

Comment:
It looks like a BC since the class is neither marked as internal or final. 

If I'm not wrong, an ideal BC layer would looks like this:
```php
private readonly ComponentPropertiesExtractor $componentPropertiesExtractor 

public function __construct(
    private string $twigTemplatesPath,
    private ComponentFactory $componentFactory,
    private Environment $twig,
    private readonly array $componentClassMap,
    ?string $anonymousDirectory = null,
    ?ComponentPropertiesExtractor $componentPropertiesExtractor = null,
) {
    parent::__construct();
    $this->anonymousDirectory = $anonymousDirectory ?? 'components';
    $this->componentPropertiesExtractor = $componentPropertiesExtractor ?? new ComponentPropertiesExtractor($this->twig);
}
```

Reply:
Good idea, it's done and pushed.